flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/hybrid/HsSpillingStrategyTestBase.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.io.network.partition.hybrid;
+
+import org.apache.flink.core.memory.MemorySegment;
+import org.apache.flink.core.memory.MemorySegmentFactory;
+import org.apache.flink.runtime.io.network.buffer.Buffer;
+import org.apache.flink.runtime.io.network.buffer.NetworkBuffer;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayDeque;
+import java.util.Deque;
+
+/** Test base for {@link HsSpillingStrategy}. */
+abstract class HsSpillingStrategyTestBase {
+    public static final int NUM_SUBPARTITIONS = 3;
+
+    public static final int MEMORY_SEGMENT_SIZE = 128;
+
+    protected HsSpillingStrategy spillStrategy;
+
+    protected TestingSpillingInfoProvider spillInfoProvider;
+
+    @BeforeEach
+    void setup() {
+        this.spillInfoProvider = new TestingSpillingInfoProvider(NUM_SUBPARTITIONS);
+    }
+
+    @Test
+    abstract void testOnBufferFinished();
+
+    @Test
+    abstract void testOnBufferConsumed();
+
+    @Test
+    abstract void testOnUsedMemoryChanged();
```

Comment:
Why do we need these abstract methods?

Reply:
removed and refactor this class into `HsSpillingStrategyTestUtils`